### PR TITLE
feat(CloudFront): Include get_distribution_config

### PR DESCRIPTION
**Pull request by jfagoagas**
#### Description

Include the `get_distribution_config` CloudFront API call.

https://docs.aws.amazon.com/cloudfront/latest/APIReference/API_GetDistributionConfig.html

### CODE REVIEW

File: moto/cloudfront/models.py [hunk 1]
```diff
@@ -274,6 +278,13 @@ def get_distribution(self, distribution_id: str) -> Tuple[Distribution, str]:
         dist.advance()
         return dist, dist.etag
 
+    def get_distribution_config(self, distribution_id: str) -> Tuple[Distribution, str]:
+        if distribution_id not in self.distributions:
+            raise NoSuchDistribution
```
Comment (bblommers) [inline moto/cloudfront/models.py:283]: Ideally this gets a separate test case, as it is distinct behaviour that we want to verify

**Review by bblommers — CHANGES_REQUESTED** [review]

File: moto/cloudfront/models.py (hunk 1 above)
Comment (jfagoagas) [reply moto/cloudfront/models.py:283]: What do you mean? I've included a test case for this new `get_distribution_config`

Comment (bblommers) [reply moto/cloudfront/models.py:283]: I meant the behaviour of calling the method with an unknown/missing identifier.

See `test_create_distribution_with_mismatched_originid` as an example - it verifies that Moto returns the correct error message when incorrect parameters are supplied. We should have a similar test case that verifies what happens when calling `get_distribution_config(Id="unknown")`

Comment (jfagoagas) [reply moto/cloudfront/models.py:283]: Totally agree, I'll review it. Thanks!

Comment (jfagoagas) [reply moto/cloudfront/models.py:283]: Included test case @bblommers.

**Review by bblommers — APPROVED** [review]
LGTM!

**Pull request merged**